### DISCUSSION TITLE
.NET Aspire Preview 3 has been released

### DISCUSSION THREAD

**Discussion by samsp-msft** [microsoft/aspire#2205] [Announcements]
.NET Aspire Preview 3 is now available! Here's a summary of what's new in this preview release:

- Dashboard
  - Now independent of the AppHost project
  - Accessibility & localization changes
  - New resource details view
  - Telemetry improvements
- Components
  - **NEW**: Azure OpenAI 
  - **NEW**: Apache Kafka 
  - **NEW**: Oracle Entity Framework 
  - **NEW**: MySQL Entity Framework 
  - **Update**: CosmosDB hosting support
  - **Update** : StackExchange.Redis now has logging
- AppModel
  - **Breaking Change**: WithServiceBinding has changed to WithEndpoint
  - **NEW**: Resource model extensions for RedisCommander and pgadmin
  - **NEW**: Resource model support for Orleans 
  - **Update**: Dapr sidecar model & support for --runtime-path option
- Deployment
  - Azure Developer CLI improvements, including support for CosmosDB & AzureSQL
- Dapr
  - IDs for Dapr sidecars no longer needs to be specified
  - First-class support for describing all Dapr components in the app model
  - Azure Developer CLI (AZD) support for deploying .NET Aspire apps that use Dapr to Azure Container Apps (ACA)

For more information, including [installation instructions](https://aka.ms/dotnet/aspire/preview3#get-net-aspire-preview-3), see the [release notes](https://aka.ms/dotnet/aspire/preview3).

**Comment by Meai1** [comment]
Seems to me like you need to add typed strings in c# because you keep adding named services, named httpclients, named aspire stuff and it's all based on magic strings. It makes no sense to me to keep going this way, why not force people to e.g use an int based enum? Surely that would be even easier to look up quickly internally.

**Comment by iphdav** [comment]
@samsp-msft Can we push the updated nuget packages please :-)

**Reply by joperezr** [reply]
Hello @iphdav, all preview 3 packages were pushed to NuGet yesterday before releasing this discussion thread. Are you seeing any issues?

**Reply by iphdav** [reply]
@joperezr I am probably confused and assume you have changed the package names or something.  I was looking in Visual Studio (nuget) and here:
https://www.nuget.org/packages/Aspire.Dashboard
But I only see preview 2 on the above link (even now).

**Reply by joperezr** [reply]
Thanks so much for clarifying. As you can see in the release notes above, one of the major items we worked on for preview 3 was how the dashboard gets launched as part of an Aspire app orchestration. Due to this, the dashboard is no longer a library, and therefore the Aspire.Dashboard package is no longer produced, meaning what you are seeing is "by design". Perhaps a better package to follow in terms of versions, would be the sdk manifest package, as that is the main entrypoint of using Aspire and unlikely to change in 8.0: https://www.nuget.org/packages/Microsoft.NET.Sdk.Aspire.Manifest-8.0.100/8.0.0-preview.3.24105.21#versions-body-tab

We can consider in the future also including a full list of packages pushed for a release in the release notes to avoid confusion.

Thanks again for the report, and please do let us know if you have any issues. Happy coding 🙂.

**Reply by iphdav** [reply]
@joperezr Thank you for the response.  I was already using the earlier dashboards (separately to Aspire) just to view my distributed logs as I was learning how to use and configure open telemetry.  I had a little 15 line Dashboard.cs class implementing IDashboardViewModelService then started it up in Program.cs like this:

```cs
var builder = Host.CreateApplicationBuilder();
builder.Services.AddHostedService(sp => new DashboardWebApplication(sp.GetRequiredService<ILogger<DashboardWebApplication>>(), sc => sc.AddSingleton<IDashboardViewModelService>(new Dashboard())));
var app = builder.Build();
app.Run();
```

I didn't mind it as a library as it meant I could do this and have it patched with library updates.  But I will have a play with what your team has done for preview 3 as the supported way of doing this going forward.
Thanks again.

**Comment by slcdlvpr** [comment]
Is preview 3 the last planned preview before general release?

**Reply by joperezr** [reply]
Hey @slcdlvpr, thanks for reaching out. While we don't have exact dates in mind yet, the current plan is to GA sometime before summer. We've been currently trying to ship roughly monthly previews to try to make sure we get all of your feedback, so I'd expect to still ship at least one or two more before the stable release.

**Comment by JamesLMiller** [comment]
Could we get a sample app of using the cosmos emulator for an Aspire app?

**Reply by davidfowl** [reply]
https://github.com/dotnet/aspire/blob/88406425f85daccec9b5cb3f22d5c8d07ebf3a8d/playground/CosmosEndToEnd/CosmosEndToEnd.AppHost/Program.cs#L6-L8

Add the **Aspire.Hosting.Azure** package to the AppHost project.

**Comment by pjbhakta-outlook** [comment]
When will ACA Jobs will be supported? Every real life application will have some backend processing as part of the whole solution

**Reply by davidfowl** [reply]
Post the first release, we'll add support. You can see the roadmap here https://github.com/dotnet/aspire/issues/1549. We're laser focused on making some fundamental pieces right.

**Comment by AmazingMuffin** [comment]
Hello,
Do you have any documentation or sample on how to connect to an Oracle database which is running in a Docker Container?
For an unknown reason, we keep getting SID not found in the tns error
Would be appreciated a lot! Thank you
